### PR TITLE
Fully Convolutional Network with Multi-Step Reinforcement Learning for Image Processing

### PR DESCRIPTION
**Pull request by AntoineTheb**

### CODE REVIEW

File: deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+pdf: "https://arxiv.org/abs/1811.04323"
+---
+
+## Introduction
+
+Reinforcement learning methods are typically used to control robots and have only recently been used for image processing purposes. The authors of the paper do the latter by assigning an agent per pixel, where the agent changes the pixel value by taking an action. They apply the method to three image processing tasks: Image denoising, image restoration and local color enhancement. They claim to achieve comparable or better performances than state-of-the-art methods.
+
+## Formally
+
+They extend A3C by assigning an agent to each pixel of the image, where each agent trains a Policy Network to determine the best action to take and a Value Network that tries to approximate the future rewards. Of course, it would be computationally impractical to actually train an agent per pixel, therefore the authors used a Fully-Convolutional Network to have the networks share the same parameters.
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md]: to have the ~networks~ **pixels** share the same parameters.

amirite?

Comment (AntoineTheb) [reply deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md]: Uhhhhhh, will re-word.

File: deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    title:   "Fully Convolutional Network with Multi-Step Reinforcement Learning for Image Processing"
+    venue:   "Arxiv"
+pdf: "https://arxiv.org/abs/1811.04323"
+---
+
+## Introduction
+
+Reinforcement learning methods are typically used to control robots and have only recently been used for image processing purposes. The authors of the paper do the latter by assigning an agent per pixel, where the agent changes the pixel value by taking an action. They apply the method to three image processing tasks: Image denoising, image restoration and local color enhancement. They claim to achieve comparable or better performances than state-of-the-art methods.
+
+## Formally
```
Comment (jhlegarreta) [inline deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md]: May be `Method` is better as a section heading.

Comment (AntoineTheb) [reply deep-learning/_posts/2019-01-22-FCN_MultiStepRL_ImageProcessing.md]: Indeed

**Comment by jhlegarreta** [comment]
Uops, I approved but later realized that we do have a `reinforcement-learning` folder as well. So may be this (and may be other RL papers reviewed) should go there?

**Comment by AntoineTheb** [comment]
@jhlegarreta Yeah I was on the fence about it. Should the "reinforcement-learning" folder include deep reinforcement learning too ? It's a mix of DL and RL. What do you guys think ? @jodoin @ppoulin91 @CharlesAuthier @lemairecarl

**Comment by lemairecarl** [comment]
I think we should have a category (folder) if it's a field of interest of someone. Example: it's really nice for me to have a `pruning-acceleration` category. I suggest you make a category for `tractography`, and just have RL as a tag.

**Comment by jhlegarreta** [comment]
IMHO, having a `tractography` folder is not (yet?) justified.

Having an application field-wise folder structure is more challenging (e.g. in this case, the paper Antoine reviewed is applied to more than one field), and may be difficult to maintain.

May be the question is whether we want to (and are able to maintain) a structure based on the applications, or rather the techniques. to me, mixing both makes things less clear.

In this regard, IMHO RL is an entity in itself, different from DL, even if it uses DL methods, networks, etc.

As an example, the `medical-imaging` folder has just one article, whereas there are many more papers that we've reviewed concerning that field. Pretty much the same with `video-analysis`.

Although I confess that having a folder where one can place and/or rapidly locate some reviews of a particular field (e.g. having to do directly with one's research subject) has its advantages.

**Comment by lemairecarl** [comment]
Let's move the discussion about categories here: https://github.com/vitalab/vitalab.github.io/issues/311

**Comment by AntoineTheb** [comment]
Soooooo, do we merge or wait on #311 to be resolved ?

**Comment by ppoulin91** [comment]
I say merge it, we'll refactor the folders later

**Pull request merged**